### DISCUSSION TITLE
Youtube API Key and Private videos

### DISCUSSION THREAD

**Discussion by xavierperez06** [lpm0073/cookiecutter-openedx-devops#23] [Q&A] [answered]
Hello, first i wanted to say thank you for your work, it's amazing.

Now, i'm trying to add videos to the LMS courses, i have a Youtube channel with some private videos, but once i added the URL it says that the video is private:

![image](https://user-images.githubusercontent.com/34634037/192358587-f95c45ad-564c-4aee-b448-8c708f72321f.png)

I went and created an API Key in [Google Cloud API Credentials](https://console.cloud.google.com/apis/credentials), and added that key to the **settings_merge.yml** file like this: **YOUTUBE_API_KEY: AIzaSyAdfylu9ynfy9DXXXXXXXXXXX**.

Then i checked the LMS config and i can see that the key is there:

![image](https://user-images.githubusercontent.com/34634037/192359445-1fc66e8b-065f-463d-adc5-036ec1f1e11e.png)

But still can see the private video, you know if i missing something? Or what can i do to make it work?

Thanks

**Comment by xavierperez06** [comment, marked as answer]
We managed to use private videos with an S3 bucket instead of Youtube.

**Comment by nadheemxab** [comment]
@xavierperez06 Can you let me know how you implemented S3 for Video ?
Do you provide an upload video option on Studio ?
